### ISSUE TITLE
Better test framework for the `model::Cfd` and `projection::Cfd` logic

### ISSUE THREAD

**Issue by da-kami** [itchysats/itchysats#3098]
At the moment we only have very few tests that cover applying events to `projection::Cfd`. This can be improved by simulating common event histories of happy path and certain edge cases. 
In `model::Cfd` we have quite a lot of tests, but they are somewhat unstructured and the `tests` module is very large, making it hard to find / add / maintain tests in there.

I think it would be great to introduce another layer of integration tests, where we define certain common event histories and test them against both `model::Cfd` and `projection::Cfd`. These tests would be as cheap as unit tests, but test both `projection::Cfd` and `model::Cfd`. 

In scope of this ticket: 
Come up with a better way to test the `apply` logic for both `model::Cfd` and `projection::Cfd`.
Write some example tests focusing on edge cases such as force-close and refund.

Out of scope:
Writing all the tests. This ticket is to create a better test framework so we can easily add tests. Once this is done please add some tickets for adding common tests so we can improve the coverage.

**Comment by bonomat** [comment]
This project is unmaintained
